### PR TITLE
Add Spawn Point Reference

### PR DESCRIPTION
**Pull request by MrMiracleMan111**
# Description

I have completed the first draft of the spawn point reference

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change which adds functionality)

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -97,6 +97,7 @@ nav:
           - Queued Games: references/queues.md
           - Scripting Introduction: tutorials/scripting_intro.md
           - Shared Data Storage: references/shared_storage.md
+          - Spawnpoints: references/spawnpoints.md
```
Comment (StanzillaManticore) [inline mkdocs.yml]:
```suggestion
          - Spawn Points: references/spawnpoints.md
```

File: src/references/spawnpoints.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+The **Farthest from Other Players** setting forces players to respawn at the spawn point farthest from all players.
+
+The **Farthest from Enemy** setting forces players to respawn at the spawn point farthest from all enemy players.
+
+The **Random Spawn Point** setting forces players to respawn at a random spawn point.
+
+## Learn More
+
+[Player](../../api/player.md) |
+[Components](../../api/components.md)
```
Comment (StanzillaManticore) [inline src/references/spawnpoints.md]: put those on the same line please

File: src/references/spawnpoints.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+![Spawn Mode Options](../img/SpawnPointReference/SpawnModeOptions.png){: .center loading="lazy"}
+
+The **None** setting forces players to respawn at the spawn point that is located at the top of the hierarchy whenever the `Spawn` command is used. Players will not automatically respawn if the **Respawn Mode** is set to **None**.
+
+The **In Place** setting forces players to respawn at the same place the player died. The player will spawn with the same rotation they had when they died.
+
+The **Round Robin** setting orders spawn points in a list from top of the heirarhcy to bottom. Players will then spawn at the spawn point at the beginning of this list. Every time a player respawns, the player will spawn at the next spawn point in the list. If a player has respawned at the last spawn point in the list, the player will then respawn at the spawn point located at the beginning of the list.
+
+| Respawn Number| Spawn point|
```
Comment (StanzillaManticore) [inline src/references/spawnpoints.md]:
```suggestion
| Respawn Number| Spawn Point|
```

**Review by StanzillaManticore — CHANGES_REQUESTED** [review]
Did `src/img/VoiceChat/editor_settings.png` slip in by accident?

**Comment by StanzillaManticore** [comment]
@PopThosePringles Can you review as well, please?

File: src/references/spawnpoints.md [hunk 4]
```diff
@@ -37,18 +49,58 @@ The **Spawn Template** property is an asset reference property that refers to an
...
 
-The **Spawn Key** property
+The **Spawn Key** property specifies what spawn points can be used. Only spawn points with matching key values to **Spawn Key** will be considered by the **Spawn Settings** object when players are spawning.
 
 #### Respawn Properties
 
-The **Respawn Mode** property
+The **Respawn Key** property specifies what spawn points can be used. Only spawn points with matching key values to **Respawn Key** will be considered by the **Spawn Settings** object when players respawn.
+
+The **Respawn Delay** property determiens how long players will have to wait before the player is respawned.
```
Comment (CommanderFoo) [inline src/references/spawnpoints.md]: Typo "property **determiens** how long"

File: src/references/spawnpoints.md [hunk 5]
```diff
@@ -37,18 +49,58 @@ The **Spawn Template** property is an asset reference property that refers to an
...
+
+There are seven settings available for the **Spawn Mode** and **Respawn Mode** properties.
+
+![Spawn Mode Options](../img/SpawnPointReference/SpawnModeOptions.png){: .center loading="lazy"}
+
+The **None** setting forces players to respawn at the spawn point that is located at the top of the hierarchy whenever the `Spawn` command is used. Players will not automatically respawn if the **Respawn Mode** is set to **None**.
+
+The **In Place** setting forces players to respawn at the same place the player died. The player will spawn with the same rotation they had when they died.
+
+The **Round Robin** setting orders spawn points in a list from top of the heirarhcy to bottom. Players will then spawn at the spawn point at the beginning of this list. Every time a player respawns, the player will spawn at the next spawn point in the list. If a player has respawned at the last spawn point in the list, the player will then respawn at the spawn point located at the beginning of the list.
```
Comment (CommanderFoo) [inline src/references/spawnpoints.md]: Typo "list from top of the **heirarhcy** to bottom"

File: src/references/spawnpoints.md [hunk 6]
```diff
@@ -37,18 +49,58 @@ The **Spawn Template** property is an asset reference property that refers to an
...
+
+| Respawn Number| Spawn point|
+|:------:|:----------:|
+| First time respawning | ![First Spawnpoint](../img/SpawnPointReference/FirstSpawnpoint.png) |
+| Second time respawning | ![Second Spawnpoint](../img/SpawnPointReference/SecondSpawnpoint.png) |
+| Third time respawning | ![Third Spawnpoint](../img/SpawnPointReference/ThirdSpawnpoint.png) |
+| Foruth time respawning | ![First Spawnpoint](../img/SpawnPointReference/FirstSpawnpoint.png) |
```
Comment (CommanderFoo) [inline src/references/spawnpoints.md]: Typo "**Foruth** time respawning"

**Review by CommanderFoo — CHANGES_REQUESTED** [review]
Few typos.

**Pull request merged**